### DISCUSSION TITLE
Why delaying "No such file or directory" Error

### DISCUSSION THREAD

**Discussion by xcdnlgd** [plotters-rs/plotters#595] [Q&A]
I just follow the quick start section in readme.
```
$ cargo run
   Compiling hello_plotters v0.1.0 (/home/xcdnlgd/study/hello_plotters)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 1.47s
     Running `target/debug/hello_plotters`
Error: BackendError(DrawingError(ImageError(IoError(Os { code: 2, kind: NotFound, message: "No such file or directory" })))
```
Turns out the error is "thrown" by 
```rust
    root.present()?;
```
I expect the error to be returned at
```rust
    let root = BitMapBackend::new("plotters-doc-data/0.png", (640, 480)).into_drawing_area();
```
Why does BitMapBackend::new() need the `path` while not using it immediately, which delays the error?
Is there any other delayed Error?

**Comment by AaronErhardt** [comment]
> Why does BitMapBackend::new() need the path while not using it immediately, which delays the error?

The reasoning is probably that you don't need the file until you draw to it. Since all `DrawingArea` types share a common interface and some might not need a path because they don't write to a file, other methods like `present()` don't take a path, which means you have to get it somewhere else. The simplest approach is to just get the path at the beginning. If you need it, it's like one or two lines to check whether you can write to that path before creating the `BitMapBackend`.

**Reply by xcdnlgd** [reply]
Thanks for your reply

> If you need it, it's like one or two lines to check whether you can write to that path before creating the BitMapBackend.

Maybe this should be added to the document.

I tried the example from https://plotters-rs.github.io/book/intro/getting_started.html

```rust
use plotters::prelude::*;

fn main() {
    let root_drawing_area = BitMapBackend::new("images/0.1.png", (1024, 768))
        .into_drawing_area();

    root_drawing_area.fill(&WHITE).unwrap();

    let mut chart = ChartBuilder::on(&root_drawing_area)
        .build_cartesian_2d(-3.14..3.14, -1.2..1.2)
        .unwrap();

    chart.draw_series(LineSeries::new(
        (-314..314).map(|x| x as f64 / 100.0).map(|x| (x, x.sin())),
        &RED
    )).unwrap();
}
```
This time, the program doesn't even panic! As if it has saved the file, which is not true.

Consider modifing the example to be something like this

```rust
    let root_drawing_area = {
        if !std::path::Path::new("image/").is_dir() {
            panic!("can't save file under 'image/'");
        }
        BitMapBackend::new("image/1.png", (1024, 768))
        .into_drawing_area()
    };
```
